torch/csrc/jit/codegen/cuda/lower_alias_memory.cpp

```diff
@@ -425,37 +548,45 @@ using AllocationInfoList = std::vector<AllocationInfoPtr>;
 //!   Outer interval marks the beginning of the loop of first write and end of
 //!   the loop of last read, both at the same loop level as the buffer
 //!   allocation.
-class BufferUseDefInfo {
+class AllocationInfoMap : private kir::IrVisitor {
  public:
   // Alias local memory if it exceeds this threshold
   static constexpr long kRegisterSizeThreshold = 1;
 
-  BufferUseDefInfo(
-      const std::vector<Expr*>& exprs,
-      BufferReuseDebugPrinter* debug_printer = nullptr)
-      : debug_printer_(debug_printer) {
-    if (debug_printer) {
-      debug_printer->buffer_info_ = this;
+  AllocationInfoMap(const std::vector<Expr*>& exprs, bool debug_print)
+      : scope_map_(exprs),
+        debug_printer_(
+            debug_print ? std::make_unique<BufferReuseDebugPrinter>()
+                        : nullptr) {
+    current_stack_.push_back(scope_map_.getGlobalScopeInfo());
+    if (debug_printer_) {
+      debug_printer_->pushScope();
+    }
+    handle(exprs);
+    if (debug_printer_) {
+      debug_printer_->popScope();
+      std::cout << debug_printer_->dumpDebugInfo(this);
     }
-    collectScopeInfo(exprs);
-    collectScopeUseDefInfo(exprs);
+    current_stack_.pop_back();
   }
 
-  //! Returns live interval info of buffer if previously
-  //!  computed.
-  c10::optional<AllocationInfoPtr> getMaybeReuseInfoFor(
-      kir::Allocate* allocate) const {
-    auto alloc_it = map_allocate_to_info_.find(allocate);
-    if (alloc_it == map_allocate_to_info_.end()) {
+  c10::optional<AllocationInfo*> getMaybeAllocationInfo(
+      const kir::Allocate* alloc) const {
+    auto it = allocation_info_map_.find(alloc);
+    if (it == allocation_info_map_.end()) {
       return c10::nullopt;
     }
-    auto alloc = alloc_it->second;
-    return alloc;
+    return it->second;
+  }
+
+  const std::unordered_map<const kir::Allocate*, AllocationInfo*>&
+  getAllocationInfoMap() const {
+    return allocation_info_map_;
   }
 
   //! Realize alias of two buffers through inner alias analysis and
   //!  keep track of the re-use.
-  void useInnerAlias(AllocationInfoPtr from, AllocationInfoPtr to) {
+  void useInnerAlias(AllocationInfo* from, AllocationInfo* to) {
```

Comment:
nit: I personally feel that `from` and `to` are confusing. I don't know who is aliasing who, and I need to look around to figure it out everytime. Would it make sense to reorder the parameter and name it as something like `(x, alias_of_x)`?

Reply:
Added comments. Hope it's better now.

Reply:
Adding comments looks good to me